DAO/MessageDAO.py

```diff
@@ -146,3 +145,18 @@ def insert_message(self, cid, uid, message, img):
         cursor.connection.commit()
 
         return message_id
+
+    def insert_reply(self, message, uid, mid, img=None):
+        cursor = self.get_cursor()
+        message_to_reply = self.get_message(mid)
+        cid = message_to_reply['cid']
```

Comment:
Looking at this from the UI's point of view. I already know the chat group I'm posting a message to and the message I'm replying to. There's no need to extract ```cid``` from message

Reply:
Right, no se me ocurrio mandar el cid del frontend